Makefile

```diff
@@ -179,24 +179,22 @@ sync-docs:
 test: test-unit
 test-all: test-unit test-ledger-mock test-race test-cover
 
-test-ledger-mock:
-	@go test -mod=readonly `go list github.com/cosmos/cosmos-sdk/crypto` -tags='cgo ledger test_ledger_mock'
+TEST_PACKAGES=./...
+TEST_TARGETS := test-unit test-unit-amino test-unit-proto test-ledger-mock test-race test-ledger test-race
 
-test-ledger: test-ledger-mock
-	@go test -mod=readonly -v `go list github.com/cosmos/cosmos-sdk/crypto` -tags='cgo ledger'
+test-unit: ARGS=-tags='cgo ledger test_ledger_mock'
+test-unit-amino: ARGS=-tags='ledger test_ledger_mock test_amino'
+test-ledger: ARGS=-tags='cgo ledger'
+test-ledger-mock: ARGS=-tags='ledger test_ledger_mock'
+test-race: ARGS=-race -tags='cgo ledger test_ledger_mock'
```

Comment:
You define `test-race` twice. Is that in purpose?

Reply:
Yes it is, see http://www.chiark.greenend.org.uk/doc/make-doc/make.html/Using-Variables.html#Target_002dspecific for further explanation.

It just sets two variables for the `test-race` rule recipe (`ARGS=-race -tags='cgo ledger test_ledger_mock'` and `TEST_PACKAGES=$(PACKAGES_NOSIMULATION)`. The result is the command `VERSION=goz-phase-1-610-g64b31d2db go test -mod=readonly -race -tags='cgo ledger test_ledger_mock' github.com/cosmos/cosmos-sdk/baseapp [snip]` being run.